R/draft.R

```diff
@@ -132,7 +132,40 @@ draft <- function(file,
   invisible(file)
 }
 
+#' List available R Markdown template in a package
+#'
+#' List the available template by name that can be used with [draft()] to create
+#' a new document for R Markdown from a package.
+#'
+#' @param package Package to list template from. Default to **rmarkdown**
+#' @param full_path Set to `TRUE` to get the full path to the available templates
+#'
+#' @return A character vector of templates name to use within [draft()]. If
+#'   `full_path = TRUE`, it returns the full path to the templates.
+#' @export
+#' @md
+#' @examples
+#' # List rmarkdown templates & create a draft
+#' available_templates()
+#'
+#' # List rticles templates
+#' \dontrun{
+#'   available_templates("rticles")
+#' }
+available_templates <- function(package = "rmarkdown", full_path = FALSE) {
+  template_folder <- pkg_file("rmarkdown", "templates", package = package)
+  if (dir_exists(template_folder)) {
+    templates <- list.files(template_folder, full.names = TRUE)
+    # exclude file in folder if any
+    templates <- templates[utils::file_test('-d', templates)]
+    if (full_path) return(templates)
+    basename(templates)
+  }
+}
+
 # List the template directories that are available for consumption.
+# This function is only used by RStudio IDE in version before 1.1.74
+# https://github.com/rstudio/rstudio/commit/01b26d1afd8e403fe1d026cba9264bf983a86173
```

Comment:
Since the commit was from five years ago (Feb 13, 2017), we can probably remove this function now. Or add a TODO item here to remind us of removing it in the future.

Reply:
Seems like this version is not supported by our support agreement anymore: https://www.rstudio.com/about/support-agreement/

However, it is still offered as download on the Old version page: https://www.rstudio.com/products/rstudio/older-versions/

So I don't know when this should be removed. I added the TODO item. 

Probably someone using a version that old will not use a recent **rmarkdown** version. but you never know 😅

Reply:
Okay, if that old version is no longer supported, I think we can just remove this function now.

If you want to be conservative, you can definitely add a `warning()` in it first, and remove it a few versions later. I guess very few people would really be affected.

Reply:
I added a warning and ask for insights the IDE team. We'll revert the warning depending on their answer for support